File: lib/data/models/user_profile.dart
import '../../core/roles.dart';

// The signed-in user as the app sees it. In production this is hydrated from
// Microsoft Entra ID; in the prototype it comes from the profile mock.

class UserProfile {
  final String id;
  final String fullName;
  final String email;
  final String position;
  final UserRole role;

  const UserProfile({
    required this.id,
    required this.fullName,
    required this.email,
    required this.position,
    required this.role,
  });
}

File: Data/Models/UserProfile.cs
using OrquestadorFrontend.Core;

namespace OrquestadorFrontend.Data.Models;

// The signed-in user as the app sees it. In production this is hydrated from
// Microsoft Entra ID; in the prototype it comes from the profile mock.
public sealed record UserProfile(
    string Id,
    string FullName,
    string Email,
    string Position,
    UserRole Role);
